lib_eio_linux/eio_linux.mli

```diff
@@ -248,4 +248,46 @@ module Low_level : sig
   (** [getaddrinfo host] returns a list of IP addresses for [host]. [host] is either a domain name or
       an ipaddress. *)
 
+  module Process : sig
+    type t
+    (** A subprocess *)
+
+    module Env : sig
+      type t
+      (** Environment variables *)
+
+      val of_list : string list -> t
+      (** Constructs environment variables from a list of strings of the form
+          ["key=value"]. *)
+    end
+
+    module Cwd : sig
+      type t =
+        | Path of Eio.Fs.dir Eio.Path.t
+        | Fd of FD.t
+     (** Working directories for processes. *)
+    end
+
+    val spawn :
+      ?env:Env.t ->
+      cwd:Cwd.t ->
+      stdin:FD.t ->
+      stdout:FD.t ->
+      stderr:FD.t ->
+      sw:Switch.t ->
+      string ->
+      string list ->
+      t
+    (** Spawns a subprocess. If the process has not finished when the switch is released, the process
```

Comment:
This diff is looking really good! Just one thing about the documentation though: there is quite a lot of useful information in the underlying [spawn ocamldoc](https://github.com/janestreet/spawn/blob/master/src/spawn.mli#L76) that is worth propagating here, now that the underlying library is no longer exposed.  For example, it answers: "what happens if [env] is None" in that documentation, which is quite important to know.